raft/src/main/java/org/apache/kafka/raft/SegmentPosition.java

```diff
@@ -14,10 +14,12 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package kafka.raft
+package org.apache.kafka.raft;
 
-import org.apache.kafka.raft.OffsetMetadata
+public record SegmentPosition(long baseOffset, int relativePosition) implements OffsetMetadata {
```

Comment:
IMO both SegmentPosition and OffsetMetadata should be in storage layer. Ideally, layers above UnifiedLog should not know that there is a concept of segment which is used to physically store data.

Happy to punt this to a different PR for later so that we can at least start with conversion of scala to java here.

Reply:
I understand your point. Today the raft module does not depend on storage. However this will likely change when we'll want to migrate the remaining raft classes out of core (`KafkaRaftManager`/`KafkaMetadataLog`). 

Let's first do the conversion then we can reconsider where to move classes.